controllers/configurationpolicy_controller.go

```diff
@@ -1371,97 +1397,171 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 		return nil, &scopedGVR, errEvent, nil
 	}
 
+	// Fetch and filter objects using provided objectSelector
 	desiredObjects := []*unstructured.Unstructured{}
+
+	// The object either has no name specified or it is templated in the object definition
+	if desiredName == "" && objectT.ObjectSelector != nil {
+		selector, err := metav1.LabelSelectorAsSelector(objectSelector)
+		if err != nil {
+			log.Error(err, "Failed to select the resources",
+				"objectSelector", fmt.Sprint(objectT.ObjectSelector.String()))
+
+			msg := fmt.Sprintf("Error parsing provided objectSelector: %v", err)
+
+			// only report this error if there wasn't another yet
+			errEvent := &objectTmplEvalEvent{
+				compliant: false,
+				reason:    "objectSelector error",
+				message:   msg,
+			}
+
+			return nil, &scopedGVR, errEvent, err
+		}
+
+		filteredObjects, err := r.DynamicWatcher.List(plc.ObjectIdentifier(), objGVK, "", selector)
```

Comment:
I'd rather perform a list query per selected namespace rather than perform client-side filtering. This should lead to less network traffic and memory usage.

Reply:
Client-side filtering (currently in the PR):
- higher controller memory usage
- one big watch on the API server
- reconciles potentially triggered when objects change that would be excluded by the namespace selector

List per relevant namespace (proposal):
- lowest controller memory usage
- multiple watches on the API server *(I think this is the one downside)*
- fewer reconciles triggered

And I think we have all of the namespace information at this point in the flow to make those requests pretty easily, is that right @dhaiducek?

Reply:
Ah, I wasn't thinking about the namespaceSelector exclusions. Yes, it ought to be straightforward to perform watches by namespace. I'll update it.